src/AnalysisEpic.cxx

```diff
@@ -0,0 +1,424 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+// Copyright (C) 2022 Gregory Matousek, Christopher Dilks
+
+#include "AnalysisEpic.h"
+
+AnalysisEpic::AnalysisEpic(TString infileName_, TString outfilePrefix_)
+  : Analysis(infileName_, outfilePrefix_)
+{};
+
+AnalysisEpic::~AnalysisEpic() {};
+
+void AnalysisEpic::Execute()
+{
+  // setup
+  Prepare();
+
+  // read EventEvaluator tree
+  TChain *chain = new TChain("events");
+  for(Int_t idx=0; idx<infiles.size(); ++idx) {
+    for(std::size_t idxF=0; idxF<infiles[idx].size(); ++idxF) {
+      // std::cout << "Adding " << infiles[idx][idxF] << " with " << inEntries[idx][idxF] << std::endl;
+      chain->Add(infiles[idx][idxF].c_str(), inEntries[idx][idxF]);
+    }
+  }
+
+  TTreeReader tr(chain);
+
+  TTreeReaderArray<Int_t> hepmcp_status(tr, "GeneratedParticles.type");
+  TTreeReaderArray<Int_t> hepmcp_PDG(tr,    "GeneratedParticles.PDG");
+  TTreeReaderArray<Float_t> hepmcp_E(tr,      "GeneratedParticles.energy");
+  TTreeReaderArray<Float_t> hepmcp_psx(tr,    "GeneratedParticles.momentum.x");
+  TTreeReaderArray<Float_t> hepmcp_psy(tr,    "GeneratedParticles.momentum.y");
+  TTreeReaderArray<Float_t> hepmcp_psz(tr,    "GeneratedParticles.momentum.z");
+
+  
+
+  // All true particles (including secondaries, etc)
+  TTreeReaderArray<Int_t> mcpart_PDG(tr,      "MCParticles.PDG");
+  TTreeReaderArray<Int_t> mcpart_genStat(tr,         "MCParticles.generatorStatus");
+  TTreeReaderArray<Int_t> mcpart_simStat(tr,         "MCParticles.simulatorStatus");
+  TTreeReaderArray<Double_t> mcpart_m(tr,         "MCParticles.mass");
+  TTreeReaderArray<Float_t> mcpart_psx(tr,       "MCParticles.momentum.x");
+  TTreeReaderArray<Float_t> mcpart_psy(tr,       "MCParticles.momentum.y");
+  TTreeReaderArray<Float_t> mcpart_psz(tr,       "MCParticles.momentum.z");
+
+
+  // Reco tracks
+  TTreeReaderArray<Int_t> tracks_type(tr,  "ReconstructedChargedParticles.type"); // needs to be made an int eventually in actual EE code
+  TTreeReaderArray<Float_t> tracks_e(tr, "ReconstructedChargedParticles.energy");
+  TTreeReaderArray<Float_t> tracks_p_x(tr, "ReconstructedChargedParticles.momentum.x");
+  TTreeReaderArray<Float_t> tracks_p_y(tr, "ReconstructedChargedParticles.momentum.y");
+  TTreeReaderArray<Float_t> tracks_p_z(tr, "ReconstructedChargedParticles.momentum.z");
+  TTreeReaderArray<Int_t> tracks_PDG(tr,  "ReconstructedChargedParticles.PDG");
+  TTreeReaderArray<Float_t> tracks_CHI2PID(tr,  "ReconstructedChargedParticles.goodnessOfPID");
```

Comment:
these are not tracks, they are reconstructed particles, which have 1-to-many relations to tracks and clusters; suggest renaming these variables... `recparts`?